libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/events/EventsTestHelper.kt

```diff
@@ -0,0 +1,466 @@
+                legCount.toLong(),
+                content["legCount"]!!.contents
+            )
+            assertEquals(
+                "stepIndex value",
+                stepIndex.toLong(),
+                content["stepIndex"]!!.contents
+            )
+            assertEquals(
+                "voiceIndex value",
+                voiceIndex.toLong(),
+                content["voiceIndex"]!!.contents
+            )
+            assertEquals(
+                "bannerIndex value",
+                bannerIndex.toLong(),
+                content["bannerIndex"]!!.contents
+            )
+            assertEquals(
+                "totalStepCount value",
+                totalStepCount.toLong(),
+                content["totalStepCount"]!!.contents
+            )
+            assertTrue(content.containsKey("appMetadata"))
+            content["appMetadata"]!!.verifyAppMetadata(appMetadata)
+        }
+    }
+
+    fun Value.verifyAppMetadata(appMetadata: AppMetadata) {
+        (contents as Map<String, Value>).let { appMetadataContent ->
+            assertEquals(
+                "appMetadata, name value",
+                appMetadata.name,
+                appMetadataContent["name"]!!.contents
+            )
+            assertEquals(
+                "appMetadata, version value",
+                appMetadata.version,
+                appMetadataContent["version"]!!.contents
+            )
+            assertEquals(
+                "appMetadata, sessionId value",
+                appMetadata.sessionId,
+                appMetadataContent["sessionId"]!!.contents
+            )
+            assertEquals(
+                "appMetadata, userId value",
+                appMetadata.userId,
+                appMetadataContent["userId"]!!.contents
+            )
+        }
+    }
+
+    fun Value.verifyTelemetryLocation(location: TelemetryLocation) {
+        (contents as Map<String, Value>).let { content ->
+            assertEquals(location.latitude, content["lat"]!!.contents)
+            assertEquals(location.longitude, content["lng"]!!.contents)
+            assertEquals(location.speed.toDouble(), content["speed"]!!.contents)
+            assertEquals(location.bearing.toDouble(), content["course"]!!.contents)
+            assertEquals(location.altitude, content["altitude"]!!.contents)
+            assertEquals(location.timestamp, content["timestamp"]!!.contents)
+            assertEquals(
+                location.horizontalAccuracy.toDouble(),
+                content["horizontalAccuracy"]!!.contents
+            )
+            assertEquals(
+                location.verticalAccuracy.toDouble(),
+                content["verticalAccuracy"]!!.contents
+            )
+        }
+    }
+
+    fun Value.verifyTelemetryLocations(array: Array<TelemetryLocation>) {
+        (contents!! as List<Value>).let { content ->
+            assertEquals(array.size, content.size)
+
+            array.forEach { locationItem ->
```

Comment:
Can't you reuse `verifyTelemetryLocation`?

Reply:
I mean sth like:
```
        (contents!! as List<Value>).let { content ->
            assertEquals(array.size, content.size)
            array.forEachIndexed { index, locationItem ->
                content[index].verifyTelemetryLocation(locationItem)
            }
          }
        }
```

Reply:
oh, ok. I wasn't sure if `Value` guarantees the order of items in an array. But seems that it is and the test will catch if it is broken